rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -44,33 +61,88 @@
  */
 public class FlowNodeUtil {
 
+    protected static final Cache<String,List<FlowNode>> executionCache = CacheBuilder.newBuilder().maximumSize(100).build();
+
+
+    // Larger cache of run data, for completed runs, keyed by flowexecution url, useful for serving info
+    // Actually can be used to serve Stage data too
+    // Because the RunExt caps the total elements returned, and this is fully realized, this is the fastest way
+    public static final Cache<String, RunExt> runData = CacheBuilder.newBuilder().maximumSize(1000).build();
+
+    static final Cache<FlowNode,String> execNodeNameCache = CacheBuilder.newBuilder().weakKeys().expireAfterAccess(1, TimeUnit.HOURS).build();
```

Comment:
There are only a couple dozen calls to `FlowNodeUtil` static methods from outside the class. A simple search-and-replace of `FlowNodeUtil.findSomething()` → `FlowNodeUtil.get().findSomething()` suffices to fix the static cache mistake. Later, if desired, you can start to thread the `FlowNodeUtil` singleton instance through different methods to reduce the number of `get()` calls, ultimately using `@Inject FlowNodeUtil util` in `Extension`s to avoid the call entirely.